sdk/identity/azure-identity-v2/pom.xml

```diff
@@ -9,10 +9,10 @@
     <relativePath>../../parents/azure-client-sdk-parent-v2</relativePath>
   </parent>
 
-  <groupId>com.azure</groupId>
+  <groupId>com.azure.v2</groupId>
```

Comment:
If the groupId is updated to `com.azure.v2`, we can drop `v2` from the artifactId.

Reply:
not sure, how eng sys will behave with two same artifact ids present in service dir.
Will defer this to a follow up PR, post checking with eng sys.